### PR TITLE
v1.8.0 | Bug fixes and save prompts

### LINKED ISSUE #188: The open project menu requires refresh if redux hasn't finished loading
If you open the project drawer quickly before firebase/redux has finished loading the projects into the state, it appears empty. The user then has to close and reopen the drawer to see the projects.

### LINKED ISSUE #256: Remove course "are you sure prompt" when no changes have been made

### LINKED ISSUE #263: Prompt to save doesn't check for changes in scene config
Steps to reproduce:

1. Open a scene.
2. Change a setting (such as sky color).
3. Close the page.

### LINKED ISSUE #366: Scenes are not being added to Collections
**Bug Description:**
Adding a Scene to a Collection fails 

**Steps To Reproduce:**
1. Create a new collection
2. Create a scene (could just be a simple box)
3. Go to Sharing Settings and add to collection
4. Enter created collection name
5. Load created collection
6. No scenes show in the collection

**Expected behavior:**
The scene would be added to the collection.  Once you open the collection, it should allow you to choose from a list of added scenes

**[Link to Video demonstrating the Bug](https://youtu.be/uaBh9YYUDDU)**

### PR DESCRIPTION
**Pull request by jasondkiesling**
- Fixed bug where you couldn't add projects to a collection
- Fixed bug where the project drawer had to be closed and reopened after loading projects
- Added prompt to save changes after changing settings
- Next course will only prompt to save if changes have been made
- Collections will be more clear on which scene is being selected when choosing a scene.
- Collections now have a default name for unnamed scenes.
- Collections will not prompt to save changes on every change.
- Box and torusknot are now double sided.

Fixes #366 
Fixes #188
Fixes #256
Fixes #263

### CODE REVIEW

**Comment by Jlu18** [comment]
In the collection, it shows the prompt every time user try to refresh the page.
OS: Windows 10
Browser: Chrome, Firefox, and Edge

**Comment by kdvalin** [comment]
If you change a setting, then change it back to whatever it was originally, the unsaved changes prompt still shows up.

**Comment by jasondkiesling** [comment]
@Jlu18 Collections seem to still be prompting to save changes whenever I switch scenes on dev.learnmyr.org.

**Comment by Jlu18** [comment]
> @Jlu18 Collections seem to still be prompting to save changes whenever I switch scenes on dev.learnmyr.org.

It should be fix on this Pull Request #375

File: src/components/structural/header/SceneConfigMenu.js [hunk 1]
```diff
@@ -81,18 +81,46 @@ class ConfigModal extends Component {
...
+                event.preventDefault();
+                event.returnValue = "You may have unsaved changes!";
+            }
+        });
+    }
+
+    buildSettingsArr = () => {
+        const sceneSettings = this.props.scene.settings;
+
+        return [sceneSettings.canFly, sceneSettings.floorColor,
```
Comment (chris-revops) [inline src/components/structural/header/SceneConfigMenu.js]: If you are just getting the values, I would try out `Object.values(scene.settings}` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Object/values

**Comment by kdvalin** [comment]
If you move the camera, then try to reload/load another scene, the browser prompts you if you would like to continue, even if no settings were changed.

More details here: https://github.com/engaging-computing/MYR/issues/263#issuecomment-577421944

**Pull request merged**